webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/assignTaxonomy/BulkAssignTaxonomyModal.js

```diff
@@ -0,0 +1,294 @@
+        type: 'success',
+        message: response.data.message,
+      })
+    );
+    dispatch(
+      APIActions.get({
+        key: API_REQUEST_KEY,
+        url: foremanUrl(HOSTS_API_PATH),
+      })
+    );
+    handleModalClose();
+  };
+
+  const handleSave = () => {
+    if (locationId) {
+      const requestBody = {
+        included: {
+          search: fetchBulkParams(),
+        },
+        location_id: locationId,
+        optimistic_import: locFixRadioChecked ? 'yes' : 'no',
+      };
+
+      dispatch(bulkAssignTaxonomy(requestBody, handleSuccess, handleError));
+    }
+
+    if (organizationId) {
+      const requestBody = {
+        included: {
+          search: fetchBulkParams(),
+        },
+        target_organization_id: organizationId,
+        optimistic_import: orgFixRadioChecked ? 'yes' : 'no',
+      };
+
+      dispatch(bulkAssignTaxonomy(requestBody, handleSuccess, handleError));
+    }
+  };
+
+  const handleOrganizationSelect = (event, selection) => {
+    setOrganizationId(selection);
+    setOrganizationSelectOpen(false);
+  };
+
+  const handleLocationSelect = (event, selection) => {
+    setLocationId(selection);
+    setLocationSelectOpen(false);
+  };
+
+  const mismatchTip = taxonomy =>
+    __(
+      `A mismatch happens if there is a resource associated with a host, such as a domain or subnet, and at the same time not associated with the ${taxonomy} you want to assign the host to. The option Fix on mismatch will add such a resource to the ${taxonomy}, and is therefore the recommended choice. The option Fail on mismatch, on the other hand, will always result in an error message. For example, reassigning a host from one ${taxonomy} to another will fail, even if there is no actual mismatch in settings.`
+    );
+
+  const modalActions = [
+    <Button
+      key="add"
+      ouiaId="bulk-assign-taxonomy-modal-add-button"
+      variant="primary"
+      onClick={handleSave}
+      isDisabled={
+        hostUpdateStatus === STATUS.PENDING ||
+        (organizationId === '' && locationId === '')
+      }
+      isLoading={hostUpdateStatus === STATUS.PENDING}
+    >
+      {__('Save')}
+    </Button>,
+    <Button
+      key="cancel"
+      ouiaId="bulk-assign-taxonomy-modal-cancel-button"
+      variant="link"
+      onClick={handleModalClose}
+    >
+      {__('Cancel')}
+    </Button>,
+  ];
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleModalClose}
+      onEscapePress={handleModalClose}
+      title={__('Change organization/ location')}
+      width="50%"
+      position="top"
+      actions={modalActions}
+      id="bulk-assign-taxonomy-modal"
+      key="bulk-assign-taxonomy-modal"
+      ouiaId="bulk-assign-taxonomy-modal"
+    >
+      <TextContent>
+        <Text ouiaId="bulk-assign-taxonomy-options">
+          {__(
+            'Select organization/ location to add systems to. This change may affect all your selected systems. Some hosts may already be in your chosen group.'
```

Comment:
"Some hosts may already be in your chosen group" was about hostgroups. 
We are not adding additional; rather, we are assigning.
We call them "hosts" and not "systems" everywhere in Satellite, so this should be consistent.


```suggestion
            'Select organization/location to assign hosts to. This change may affect all your selected hosts.'
```

Reply:
Those messages come from the provided mock page. 
I could use your suggestion as well.

Reply:
I vote for my suggestion :innocent: but will also tag @MariSvirik in case she thinks more argument is needed :)

Reply:
:+1: for what @jeremylenz said because Foreman uses hosts, not systems. (Katello used to have systems, but thankfully it's now aligned on hosts).